antarest/study/business/binding_constraint_management.py

```diff
@@ -912,13 +930,19 @@ def _replace_matrices_according_to_frequency_and_version(
             BindingConstraintFrequency.DAILY.value: default_bc_weekly_daily_87,
             BindingConstraintFrequency.WEEKLY.value: default_bc_weekly_daily_87,
         }[data.time_step].tolist()
-        for term in TERM_MATRICES:
+        for term in [m.value for m in TermMatrices]:
             if term not in args:
                 args[term] = matrix
     return args
 
 
-def check_attributes_coherence(data: t.Union[ConstraintCreation, ConstraintInput], study_version: int) -> None:
+def check_attributes_coherence(
+    data: t.Union[ConstraintCreation, ConstraintInput],
+    study_version: int,
+    existing_operator: t.Optional[BindingConstraintOperator] = None,
+) -> None:
+    update_operator = data.operator or existing_operator
```

Comment:
The way you've handled it is disturbing IMO. You should not need to  add this line. When calling this method for creation, give data.operator or the default operator we use (tI think it's less). When calling for an update, use data.operator or existing_constraint.operator. So you'll always have an operator here, it won't be optional and it will simplify your code

Reply:
I didn't understand what you mean.
Can you be more specific? 

This function is called both on creation and update.